### PR TITLE
DO NOT Merge: Add commit confirmed plus commit message support

### PR DESCRIPTION
**Pull request by ktbyers**

### CODE REVIEW

**Comment by ktbyers** [comment]
Resubmitted due to some merge conflicts...see additional notes here: https://github.com/napalm-automation/napalm/pull/544

File: napalm/junos/junos.py [hunk 1]
```diff
@@ -234,8 +234,11 @@ def compare_config(self):
         else:
             return diff.strip()
 
-    def commit_config(self):
+    def commit_config(self, confirmed=None, message=None):
         """Commit configuration."""
+        if confirmed is not None:
```
Comment (mirceaulinic) [inline napalm/junos/junos.py]: Junos supports commit confirmed (and message).

File: napalm/iosxr/iosxr.py [hunk 2]
```diff
@@ -145,7 +145,10 @@ def compare_config(self):
         else:
             return self.device.compare_config().strip()
 
-    def commit_config(self):
+    def commit_config(self, confirmed=None, message=None):
+        if confirmed is not None:
```
Comment (mirceaulinic) [inline napalm/iosxr/iosxr.py]: IOS-XR too (also message).

**Comment by ktbyers** [comment]
@mirceaulinic Cisco IOS should be ready. I tested that when I implemented it.

**Comment by ktbyers** [comment]
@mirceaulinic Let me know when you add this into other platforms and I will regression test it in my lab environment.

We should also regression test it when it gets merged into develop branch (which I should be able to do also).

File: napalm/base/base.py [hunk 3]
```diff
@@ -178,12 +178,16 @@ def compare_config(self):
         """
         raise NotImplementedError
 
-    def commit_config(self):
+    def commit_config(self, confirmed=None, message=None):
```
Comment (dbarrosop) [inline napalm/base/base.py]: We should document what "confirmed" is supposed to be.

**Comment by dbarrosop** [comment]
I think this whole functionality requires two extra methods:

1. `has_pending_commit()` that returns whether if the system has a commit pending confirmation.
2. `revert_commit()` basically to shortcircuit the timer and revert immediately.

File: napalm/base/base.py (hunk 3 above)
Comment (dbarrosop) [reply napalm/base/base.py]: I also really think we shouldn't name this variable `confirmed`. It is a terrible name; `revert_in` or `timer` are way better names.

Comment (ktbyers) [reply napalm/base/base.py]: `confirmed` is logical and is the most widely adopted name for this function. Basically saying you have to `confirm` this commit in how many minutes.

I think we will probably just add more confusion by naming it something else.

But agreed...it needs documented.

**Comment by ktbyers** [comment]
Added two methods per @dbarrosop comment (although I changed the name of one of them). Implemented and tested them on Cisco IOS.

File: napalm/base/base.py (hunk 3 above)
Comment (sdanelli) [reply napalm/base/base.py]: +1 on `revert_in`, describes the intent better. `confirmed` nomenclature maybe better known by network engineers, but I wouldn't cater to them exclusively with this library, you may have engineers without that background.

**Comment by dbarrosop** [comment]
Awesome! :)

**Comment by ktbyers** [comment]
@dbarrosop @mirceaulinic Did some more research and review (which also helped me remember past research):

1. IOS - Should be able to do reasonable system-wide solution.

2. NX-OS - Does not have any reasonable solution (AFAIK). The only thing I see is a kludge that we shouldn't do. This is what I found in past research. We can ask in Cisco channel or NAPALM channel and see if anyone know of a reasonable solution.

3. EOS - I think there is support, but a specific version is required. I believe I had this discussion with @bewing previously.

4. Junos - should be fine

5. IOS-XR / can you commit-confirm in a different SSH session on IOS-XR? I wasn't seeing a good way of doing this in quick testing. @mirceaulinic

**Comment by dbarrosop** [comment]
We need to confirm 5. @mirceaulinic is that something you can look into?

**Comment by ktbyers** [comment]
Closing...don't think it makes sense to keep this PR open any longer.

**Pull request closed without merging**